### PR TITLE
testnet cmd: add config option

### PR DESCRIPTION
**Pull request by gregdhill**
Option to explicitly provide the `-config` in the `testnet` command, closing #3160.

* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [ ] ~~Wrote tests~~
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by melekes** [comment]
Why don't the ability to provide the whole config (`--config`)?

**Comment by gregdhill** [comment]
@melekes I haven't defined the whole config due to the initial specification in the linked issue.

**Comment by melekes** [comment]
See https://github.com/tendermint/tendermint/issues/3160#issuecomment-483538923. Sorry about that. Shouldn't be too hard to refactor this PR though

File: cmd/tendermint/commands/testnet.go [hunk 1]
```diff
@@ -73,6 +77,23 @@ Example:
...
 	config := cfg.DefaultConfig()
+
+	// overwrite default config if set and valid
+	if configFile != "" {
+		base := &cfg.Config{}
+		viper.SetConfigFile(configFile)
+		if err := viper.ReadInConfig(); err != nil {
+			return err
+		}
+		if err := viper.Unmarshal(base); err != nil {
```
Comment (melekes) [inline cmd/tendermint/commands/testnet.go]: maybe use `config` here? that's what we do in cmd/tendermint/root.go#ParseConfig

Comment (gregdhill) [reply cmd/tendermint/commands/testnet.go]: Done :)

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 ### IMPROVEMENTS:
 - [rpc] [\#3534](https://github.com/tendermint/tendermint/pull/3534) Add support for batched requests/responses in JSON RPC
+- [cli] [\#3160](https://github.com/tendermint/tendermint/issues/3160) Add base config option to testnet cmd
```
Comment (melekes) [inline CHANGELOG_PENDING.md]:
```suggestion
- [cli] [\#3160](https://github.com/tendermint/tendermint/issues/3160) Add `--config=<path-to-config.toml>` option to `testnet` cmd
```

File: cmd/tendermint/commands/testnet.go [hunk 3]
```diff
@@ -36,6 +38,8 @@ const (
 func init() {
 	TestnetFilesCmd.Flags().IntVar(&nValidators, "v", 4,
 		"Number of validators to initialize the testnet with")
+	TestnetFilesCmd.Flags().StringVar(&configFile, "config", "",
+		"Use this file as the base configuration")
```
Comment (melekes) [inline cmd/tendermint/commands/testnet.go]:
```suggestion
		"Config file to use (note some options may be overwritten)")
```

**Pull request merged**